src/controllers/jobSeeker.ts

```diff
@@ -1,46 +1,107 @@
+  NotFoundError,
+  UnauthorizedError,
+  InternalServerError,
+  BadRequestError,
+} from '../helpers/apiError'
 import JobSeeker from '../entities/JobSeeker.postgres'
+import Credential from '../entities/Credential.postgres'
+import { getConnection, UpdateDateColumn } from 'typeorm'
 
-export const getJobSeeker = async (
+// Auth Controllers for job seeker
+export const jobSeekerLocalLogin = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  passport.authenticate('local', (error, jobSeeker, info) => {
+    if (error) {
+      return next(new InternalServerError())
+    }
+    if (!jobSeeker) {
+      if (info.message === 'Invalid email or password') {
+        return next(new UnauthorizedError(info.message))
+      }
+      return next(new NotFoundError(info.message))
+    }
+    res.status(200).send(jobSeeker)
+  })(req, res, next)
+}
+
+// Register Job Seeker
+export const createJobSeeker = async (
   req: Request,
   res: Response,
   next: NextFunction
 ) => {
   try {
-    const user = await JobSeeker.find()
-    res.send(user)
+    const { info, credential } = req.body
+    const exists = await Credential.findOne({
+      where: { email: credential.email },
+    })
+
+    if (exists) {
+      next(new BadRequestError(`Email ${credential.email} already exists`))
+    }
+    credential.password = await bcrypt.hash(credential.password, 8)
+    const newCredential = Credential.create({ ...credential })
+    const newJobSeeker = JobSeeker.create({
+      ...info,
+      credentials: newCredential,
+    })
+
+    await JobSeeker.save(newJobSeeker)
+
+    res.send('success')
   } catch (error) {
-    console.log(error)
+    next(new InternalServerError(error.message))
   }
 }
 
-export const getJobSeekerByFirstName = async (
+// getting jobSeeker based on matched credentials
+export const getJobSeeker = async (
   req: Request,
   res: Response,
   next: NextFunction
 ) => {
   try {
-    const name = req.params.name
-    console.log(name)
-    const user = await JobSeeker.getByFirstName(name)
-    res.send(user)
+    const user = await JobSeeker.find({ relations: ['credentials'] })
+    res.json(user)
   } catch (error) {
     console.log(error)
   }
 }
 
-export const createJobSeeker = async (
+// JobSeek update
+
+export const updateJobSeeker = async (
   req: Request,
   res: Response,
   next: NextFunction
 ) => {
   try {
-    const user = { ...req.body } as JobSeeker
-    await JobSeeker.create(user).save()
-
-    res.send('success')
+    const update = req.body
+    console.log('update get body request', update)
+    const jobSeekerId = req.params.id
+    console.log('Jobseeker ID ', jobSeekerId)
+    await getConnection()
+      .createQueryBuilder()
+      .update(JobSeeker)
+      .set({
+        firstName: update.firstName,
+        lastName: update.lastName,
+        contact: update.contact,
+        relocate: update.relocate,
+        seniority: update.seniority,
+        startingDate: update.startingDate,
```

Comment:
Can you add a check whether if the update includes all of this properties. Because not all the time the user wants to update the whole profile and only update what is sent in the form

Reply:
should i add it here or in entity ?